src/components/connectAdvanced.js

```diff
@@ -140,7 +146,11 @@ export default function connectAdvanced(
 
       static getDerivedStateFromProps(props, state) {
         let ret = null
-        if (state.lastNotify !== state.notifyNestedSubs) {
+        // this next check only should trigger if gDSFP reacts to state changes
+        // in React 16.3, it doesn't. react-lifecycles-compat matches React 16.3 behavior
+        // if a future version of react-lifecycles-compat DOES trigger on state changes
+        // the isOldReact() call must be removed
+        if (!isOldReact() && state.lastNotify !== state.notifyNestedSubs) {
```

Comment:
No reason to do the string splitting every time this runs.  We can just do it once when `connect()` is called, or even once at the top of the file.

Reply:
duh. fixing :)